src/compiler/checker.ts

```diff
@@ -31261,12 +31266,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const paramSymbol = createSymbol(
                 SymbolFlags.FunctionScopedVariable | (isOptional && !isRestParam ? SymbolFlags.Optional : 0),
                 paramName || `arg${i}` as __String,
+                isRestParam ? CheckFlags.RestParameter : isOptional ? CheckFlags.OptionalParameter : 0,

```

Comment:
adding `CheckFlags.RestParameter` here is based on the similar change from https://github.com/microsoft/TypeScript/pull/55625